pkg/operator/target_config_reconciler.go

```diff
@@ -0,0 +1,364 @@
+	allNamespaces, err := kubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.ErrorS(err, "error listing namespaces")
+		return nil
+	}
+	excludedNamespaces := []string{"kube-system"}
+	for _, ns := range allNamespaces.Items {
+		if strings.HasPrefix(ns.Name, "openshift-") {
+			excludedNamespaces = append(excludedNamespaces, ns.Name)
+		}
+	}
+
+	c := &TargetConfigReconciler{
+		ctx:                      ctx,
+		operatorClient:           operatorConfigClient,
+		secondarySchedulerClient: secondarySchedulerClient,
+		kubeClient:               kubeClient,
+		osrClient:                osrClient,
+		dynamicClient:            dynamicClient,
+		eventRecorder:            eventRecorder,
+		queue:                    workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "TargetConfigReconciler"),
+		excludedNamespaces:       excludedNamespaces,
+		targetImagePullSpec:      targetImagePullSpec,
+	}
+	operatorClientInformer.Informer().AddEventHandler(c.eventHandler())
+
+	return c
+}
+
+func (c TargetConfigReconciler) sync() error {
+	secondaryScheduler, err := c.operatorClient.SecondarySchedulers(operatorclient.OperatorNamespace).Get(c.ctx, operatorclient.OperatorConfigName, metav1.GetOptions{})
+	if err != nil {
+		klog.ErrorS(err, "unable to get operator configuration", "namespace", operatorclient.OperatorNamespace, "secondary-scheduler", operatorclient.OperatorConfigName)
+		return err
+	}
+
+	forceDeployment := false
+	_, forceDeployment, err = c.manageConfigMap(secondaryScheduler)
+	if err != nil {
+		return err
+	}
+
+	// if _, _, err := c.manageServiceAccount(secondaryScheduler); err != nil {
+	// 	return err
+	// }
+
+	if _, _, err := c.manageClusterRoleBinding(secondaryScheduler); err != nil {
+		return err
+	}
+
+	deployment, _, err := c.manageDeployment(secondaryScheduler, forceDeployment)
+	if err != nil {
+		return err
+	}
+
+	_, _, err = v1helpers.UpdateStatus(c.secondarySchedulerClient, func(status *operatorv1.OperatorStatus) error {
+		resourcemerge.SetDeploymentGeneration(&status.Generations, deployment)
+		return nil
+	})
+	return err
+}
+
+func (c *TargetConfigReconciler) manageConfigMap(secondaryScheduler *secondaryschedulersv1.SecondaryScheduler) (*v1.ConfigMap, bool, error) {
+	required := resourceread.ReadConfigMapV1OrDie(v410_00_assets.MustAsset("v4.1.0/secondary-scheduler/configmap.yaml"))
+	required.Name = secondaryScheduler.Name
+	required.Namespace = secondaryScheduler.Namespace
+	required.OwnerReferences = []metav1.OwnerReference{
+		{
+			APIVersion: "v1",
+			Kind:       "SecondaryScheduler",
+			Name:       secondaryScheduler.Name,
+			UID:        secondaryScheduler.UID,
+		},
+	}
+
+	if secondaryScheduler.Spec.SchedulerConfig == secondaryschedulersv1.CustomizedConfig {
+		yamlFile, err := ioutil.ReadFile(SecondarySchedulerCommand[2])
+		if err != nil {
+			return nil, false, err
+		}
+		required.Data = map[string]string{"config.yaml": string(yamlFile)}
```

Comment:
It may be more accessible if custom configs were read from a user-created configMap, which is copied to one managed by the operator. Something similar to the process we had for Policy configs in the old scheduler operator: https://docs.openshift.com/container-platform/4.1/nodes/scheduling/nodes-scheduler-default.html